client/src/app/shared/components/head-bar/head-bar.component.ts

```diff
@@ -78,33 +113,104 @@ export class HeadBarComponent implements OnInit {
     public plusButtonClicked = new EventEmitter<boolean>();
 
     /**
-     * Emit a signal to the parent of an item in the menuList was selected.
+     * Sends a signal if a detail view should be edited or editing should be canceled
+     */
+    @Output()
+    public editEvent = new EventEmitter<boolean>();
+
+    /**
+     * Sends a signal if a detail view should be saved
      */
     @Output()
-    public ellipsisMenuItem = new EventEmitter<any>();
+    public saveEvent = new EventEmitter<boolean>();
 
     /**
      * Empty constructor
      */
-    public constructor() {}
+    public constructor(
+        public vp: ViewportService,
+        private scrollDispatcher: ScrollDispatcher,
+        private ngZone: NgZone,
+        private menu: MainMenuService,
+        private router: Router,
+        private route: ActivatedRoute,
+        private location: Location
+    ) {}
 
     /**
-     * empty onInit
+     * Emits a signal to the parent if
      */
-    public ngOnInit(): void {}
+    public clickPlusButton(): void {
+        this.plusButtonClicked.emit(true);
+    }
 
     /**
-     * Emits a signal to the parent if an item in the menu was clicked.
-     * @param item
+     * Clicking the burger-menu-icon should toggle the menu
      */
-    public clickMenu(item: any): void {
-        this.ellipsisMenuItem.emit(item);
+    public clickHamburgerMenu(): void {
+        this.menu.toggleMenu();
     }
 
     /**
-     * Emits a signal to the parent if
+     * Toggle edit mode and send a signal to listeners
      */
-    public clickPlusButton(): void {
-        this.plusButtonClicked.emit(true);
+    public toggleEditMode(): void {
+        this.editEvent.next(!this.editMode);
+    }
+
+    /**
+     * Send a save signal and set edit mode
+     */
+    public save(): void {
+        if (this.editMode) {
+            this.saveEvent.next(true);
+        }
+    }
+
+    /**
+     * Exits the view to return to the previous page or
+     * visit the parent view again.
+     *
+     * TODO: for some view, like tags or legal notice, the parent
+     *       view might not be the correct target.
+     */
+    public exitView(): void {
```

Comment:
Renamed. Todo was already outdated. Components can decide weather they want to navigate to the parent or just use `location.back();`
I want to avoid routing in the head bar, so this should be sufficient for must cases

Reply:
Renamed. Todo was already outdated. Components can decide weather they want to navigate to the parent or just use `location.back();`
I want to avoid routing in the head bar, so this should be sufficient for must cases